Content.Server/ColorShift/ColorShiftSystem.cs

```diff
@@ -0,0 +1,62 @@
+using Content.Server.DoAfter;
+using Content.Server.Humanoid;
+using Content.Shared.ColorShift;
+using Content.Shared.DoAfter;
+using Content.Shared.Humanoid;
+using Content.Shared.Humanoid.Markings;
+using Content.Shared.Mind;
+using Content.Shared.Mind.Components;
+
+namespace Content.Server.ColorShift;
+
+/// <summary>
+/// This handles the special ability for slimes to color-shift to different colors.
+/// </summary>
+public sealed class ColorShiftSystem : SharedColorShiftSystem
+{
+    [Dependency] private readonly HumanoidAppearanceSystem _appearanceSystem = default!;
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+    [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ColorShifterComponent, PleaseHueShiftNetworkMessage>(OnMessageReceive);
+        SubscribeLocalEvent<HumanoidAppearanceComponent, HueShiftDoAfterEvent>(OnDoAfter);
+    }
+
+    private void OnDoAfter(Entity<HumanoidAppearanceComponent> ent, ref HueShiftDoAfterEvent args)
+    {
+        if (args.Cancelled)
+            return;
+
+        var color = args.NewColor;
+        var colorList = new List<Color> { color }; // needed for markings
+        _appearanceSystem.SetSkinColor(ent.Owner, color);
+        // Hardcoding it to zero is a bandaid until someone can figure out how to loop through all possible hair marking slots and set the color there
```

Comment:
You can use `HumanoidAppearanceComponent.MarkingSet::TryGetCategory(MarkingsCategory, out IReadOnlyList<Marking>?)` to get a list of all markings and then you can get an index from there.